src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -0,0 +1,314 @@
+                   File.Exists(path) &&
+                   File.Exists($"{path}.config");
+        }
+
+        /// <summary>
+        /// Look for Visual Studio
+        /// </summary>
+        /// <param name="processName"></param>
+        /// <param name="processName2"></param>
+        /// <param name="runningInVisualStudio"></param>
+        /// <returns></returns>
+        private static string FindVisualStudio(string[] processNames, out bool runningInVisualStudio)
+        {
+            runningInVisualStudio = false;
+
+            // Check to see if we're running inside of Visual Studio
+            foreach (var process in processNames)
+            {
+                if (IsProcessInList(process, s_visualStudioProcess))
+                {
+                    runningInVisualStudio = true;
+
+                    // This assumes running from VS\Common7\IDE\devenv.exe.
+                    return FileUtilities.GetFolderAbove(process, 3);
+                }
+            }
+
+            // VSInstallDir is set from the Developer Command Prompt
+            var vsInstallDir = Environment.GetEnvironmentVariable("VSINSTALLDIR");
+            var vsVersion = Environment.GetEnvironmentVariable("VisualStudioVersion");
+
+            if (!string.IsNullOrEmpty(vsInstallDir) &&
+                !string.IsNullOrEmpty(vsVersion) &&
+                vsVersion == "15.0" &&
+                Directory.Exists(vsInstallDir))
+            {
+                return vsInstallDir;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Returns true if processName appears in the processList
+        /// </summary>
+        /// <param name="processName">Name of the process</param>
+        /// <param name="processList">List of processes to check</param>
+        /// <returns></returns>
+        private static bool IsProcessInList(string processName, string[] processList)
+        {
+            return processList.Any(s => processName?.IndexOf(s, StringComparison.InvariantCultureIgnoreCase) >= 0);
+        }
+
+        private static string GetProcessFromCommandLine()
+        {
+            return Path.GetFileNameWithoutExtension(Environment.GetCommandLineArgs()[0]);
+        }
+
+        private static string GetProcessFromRunningProcess()
+        {
+            return Process.GetCurrentProcess().ProcessName;
+        }
+
+        private static string GetExecutingAssmblyPath()
+        {
+            return FileUtilities.ExecutingAssemblyPath;
+        }
+
+        private static string GetCurrentDirectory()
+        {
+            return Directory.GetCurrentDirectory();
+        }
+
+        /// <summary>
+        /// Resets the current singleton instance (for testing).
+        /// </summary>
+        internal static void ResetInstance(Func<string> getProcessFromCommandLine = null, Func<string> getProcessFromRunningProcess = null, Func<string> getExecutingAssmblyPath =  null, Func<string> getCurrentDirectory = null)
```

Comment:
Is there a convention for marking test only methods? In some other class (I think the whitelisted property functions) I saw the test methods prepended with "_TestOnly"